### PR TITLE
feat: support release zetasql package from mac os

### PR DESCRIPTION
**Pull request by jingchen2222**
* **Please check if the PR fulfills these requirements**
    - The commit message follows our guidelines
    - Tests for the changes have been added (for bug fixes / features)
    - Docs have been added / updated (for bug fixes / features)
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
Resolve  https://github.com/4paradigm/HybridSE/issues/60
feat: support release zetasql package in MacOS

* **What is the new behavior (if this is a feature change)?**
  - support pack zetasql in macos
  - support release package from cicd

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -196,3 +200,51 @@ jobs:
...
+        run: |
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          VERSION=$(echo $VERSION | sed -e 's/^v//')
+          echo "TAG=$VERSION" >> $GITHUB_ENV
+      
+      - name: pack libzetasql
+        run: |
+          ./pack_zetasql.sh
+        env:
+          TAG: ${{ env.TAG }}
```
Comment (aceforeverd) [inline .github/workflows/main.yml]: this `TAG` is not same as the release tag: `1.0.0` vs `v1.0.0`

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -196,3 +200,51 @@ jobs:
...
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  mac-release:
+    runs-on: macos-latest
+    if: startsWith(github.ref, 'refs/tags/v')
+    timeout-minutes: 120
+    env:
+      build_argv: '--features=-supports_dynamic_linker '
+      bazel_version: '3.7.2'
```
Comment (aceforeverd) [inline .github/workflows/main.yml]: reason ?

File: pack_zetasql.sh [hunk 3]
```diff
@@ -7,75 +7,132 @@ set -eE
...
-
-
-tar czf "libzetasql-$VERSION.tar.gz" "libzetasql-$VERSION"/
+find zetasql -type f -iname "*.h" -exec ${INSTALL_BIN} -D {} "$PREFIX"/include/{} \;
+
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]
+then
+    echo 'create libzetasql.a' >> libzetasql.mri
+    find tmp-lib/ -iname "*.a" -type f -exec bash -c 'echo "addlib $0" >> libzetasql.mri' {} \;
```
Comment (aceforeverd) [inline pack_zetasql.sh:121]: libtool should preferred here as well, forgot why I don't use it though.

File: .github/workflows/main.yml (hunk 2 above)
Comment (jingchen2222) [reply .github/workflows/main.yml]: > reason ?

zetasql need  bazel 3.7.2+  to build absl in macos

**Pull request merged**